content/300-guides/100-performance-and-optimization/090-metrics.mdx

```diff
@@ -0,0 +1,497 @@
+
+Prisma Client provides the following metrics:
+
+- Counters (always increase):
+
+  - `query_total_queries`: The total number of database queries executed.
+  - `query_total_operations`: The total number of Prisma operations executed.
+    - A Prisma operation is a statement like `let user = prisma.user(include: {posts: true}})`, which Prisma Client translates into one or more SQL queries.
+    - The value returned by `query_total_operations` can be less than `query_total_queries`, because some operations create multiple queries.
+
+- Gauges (can increase or decrease):
+
+  - `pool_active_connections`: The number of active connections in use.
+  - `pool_idle_connections`: The number of connections that are not being used.
+  - `pool_wait_count`: The number of workers waiting for a connection.
+  - `query_active_transactions`: The number of active transactions.
+
+- Histograms (metrics data divided into a collection of values; we call each container in the collection a "bucket"):
+
+  - `pool_wait_duration_ms`: The wait time for a worker to get a connection.
+  - `query_total_elapsed_time_ms`: The execution time for all database queries executed.
+  - `query_operation_total_elapsed_time_ms`: The execution time for all operations.
+
+You can [add global labels to your metrics data](#global-labels) to help you group and separate your metrics, for example by infrastructure region or server.
+
+## Prerequisites
+
+To use Prisma metrics, you must do the following:
+
+1. [Install the appropriate dependencies](#step-1-install-up-to-date-prisma-dependencies).
+1. [Enable the `metrics` feature flag in your Prisma schema file](#step-2-enable-the-feature-flag-in-the-prisma-schema-file).
+
+### Step 1. Install up-to-date Prisma dependencies
+
+Use version `3.15.0` or higher of the `prisma` and `@prisma/client` npm packages.
+
+```terminal
+npm install prisma@latest --save-dev
+npm install @prisma/client@latest --save
+```
+
+### Step 2: Enable the feature flag in the Prisma schema file
+
+In the `generator` block of your `schema.prisma` file, enable the `metrics` feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["metrics"]
+}
+```
+
+## Retrieve metrics in JSON format
+
+When you retrieve metrics in JSON format, you can use them in the format they are returned, or [send them to StatSD](#use-prisma-metrics-with-statsd) to visualize how they change over time.
+
+<Admonition type="info">
+
+**Histograms in JSON data**<br /><br />
+
+Each histogram "bucket" has two values. The first one is the upper bound of the bucket, and the second one is the count (the number of data values that fall into that bucket). In the following example, there are two instances of a query returning a value between 11 and 20, and five instances of a query returning a value between 21 and 30:
+
+```json
+...
+[20, 2],
+[30, 5],
+...
+```
+
+</Admonition>
```

Comment:
The "info" Admonition visually formats this was it was something very important:
<img width="535" alt="image" src="https://user-images.githubusercontent.com/183673/172251217-8d587b2d-2833-41b7-9f31-39c5012e6503.png">

Reply:
This might be a weird bug, the other info Admonition does not get the bold text: 
<img width="599" alt="image" src="https://user-images.githubusercontent.com/183673/172251720-6c13784f-fda3-4fa9-80f1-5b970d1beb7e.png">

Reply:
Good point. There appears to be a bug with admonitions and code samples. To work around this, I've moved the admonition content to below the code sample. I don't love this solution, but I think it's the least-worst situation we can have. I could also recast the histograms example, by removing the code block in it, but that makes the explanation less clear.

Reply:
Make sure to create an issue for the webdev people so this can be fixed - I think no one before could isolate the problem that well. Might also be worth creating a fake PR that shows the problem in multiple code blocks so it is reproducible.

Reply:
Will do!